src/main/java/com/cloudbees/jenkins/plugins/bitbucket/impl/notifier/BitbucketBuildStatusNotifications.java

```diff
@@ -177,10 +177,14 @@ private static void createStatus(@NonNull Run<?, ?> build,
         if (state != null) {
             BitbucketDefaulNotifier notifier = new BitbucketDefaulNotifier(client);
             String notificationKey = DigestUtils.md5Hex(key);
+            String parentKey = getParentKey(build);
+            String notificationParentKey = DigestUtils.md5Hex(parentKey);
             if (context.useReadableNotificationIds() && !isCloud) {
                 notificationKey = key.replace(' ', '_').toUpperCase();
+                notificationParentKey =  parentKey.replace(' ', '_').toUpperCase();
             }
-            BitbucketBuildStatus buildStatus = new BitbucketBuildStatus(hash, statusDescription, state, url, notificationKey, name, refName);
+
+            BitbucketBuildStatus buildStatus = new BitbucketBuildStatus(hash, statusDescription, state, url, notificationKey, notificationParentKey, name, refName);
             buildStatus.setBuildDuration(build.getDuration());
```

Comment:
set here `buildStatus.setParent`.
Always null expcept when useReadableNotificationIds is enable and it not cloud